hrm-service/src/case/case-sql.ts

```diff
@@ -40,10 +40,16 @@ const generateOrderByClause = (
   } else return '';
 };
 
-const SELECT_CONTACTS = `SELECT c.*,
-COALESCE(jsonb_agg(DISTINCT r.*) FILTER (WHERE r.id IS NOT NULL), '[]') AS "csamReports"
-FROM "Contacts" c LEFT JOIN "CSAMReports" r ON c."id" = r."contactId" WHERE c."caseId" = "cases".id
-GROUP BY c.id`;
+const SELECT_CONTACTS = `SELECT COALESCE(jsonb_agg(DISTINCT contacts.*) FILTER (WHERE contacts."caseId" IS NOT NULL), '[]') AS "connectedContacts"
+FROM (
+  SELECT
+    c.*,
+    COALESCE(jsonb_agg(DISTINCT r.*) FILTER (WHERE r.id IS NOT NULL), '[]') AS "csamReports"
+  FROM "Contacts" c 
+  LEFT JOIN "CSAMReports" r ON c."id" = r."contactId" 
+  WHERE c."caseId" = "cases".id
+  GROUP BY c.id
+) AS contacts WHERE contacts."caseId" = cases.id`;
```

Comment:
As we talked via Slack, I wonder if this `WHERE contacts."caseId" = cases.id` clause is needed in both queries.